actors/builtin/multisig/multisig_test.go

```diff
@@ -75,10 +78,208 @@ func TestConstruction(t *testing.T) {
 	})
 }
 
-func newAddr(t *testing.T, id uint64) addr.Address {
+func TestPropose(t *testing.T) {
+	actor := msActorHarness{multisig.MultiSigActor{}, t}
+
+	receiver := newIDAddr(t, 100)
+	anne := newSecpAddr(t, 101)
+	bob := newSecpAddr(t, 102)
+	chuck := newSecpAddr(t, 103)
+
+	builder := mock.NewBuilder(context.Background(), t, receiver).WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple propose", func(t *testing.T) {
+		const unlockDuration = int64(0)
+		const numApprovals = int64(2)
+		const method = int64(0)
+		const value = int64(10)
+
+		var nilParams = abi.MethodParams{}
+		var signers = []addr.Address{anne, bob}
+
+		rt := builder.Build()
+
+		actor.mustConstructMultisigActor(rt, numApprovals, unlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.mustPropose(rt, chuck, value, method, nilParams)
+
+		actor.mustContainTransactions(rt, multisig.MultiSigTransaction{
+			To:       chuck,
+			Value:    abi.NewTokenAmount(value),
+			Method:   abi.MethodNum(method),
+			Params:   nilParams,
+			Approved: []addr.Address{anne},
+		})
+	})
+
+	t.Run("propose with threshold met", func(t *testing.T) {
+		const unlockDuration = int64(0)
+		const numApprovals = int64(1)
+		const method = int64(0)
+		const value = int64(10)
+
+		var nilParams = abi.MethodParams{}
+		var signers = []addr.Address{anne, bob}
+
+		rt := builder.WithBalance(abi.NewTokenAmount(20), abi.NewTokenAmount(0)).Build()
+
+		actor.mustConstructMultisigActor(rt, numApprovals, unlockDuration, signers...)
+
+		rt.ExpectSend(chuck, abi.MethodNum(method), nilParams, abi.NewTokenAmount(value), nil, 0)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.mustPropose(rt, chuck, value, method, nilParams)
+
+		actor.mustContainTransactions(rt)
+	})
+
+	t.Run("fail propose with threshold met and insufficient balance", func(t *testing.T) {
+		const unlockDuration = int64(0)
+		const numApprovals = int64(1)
+		const method = int64(0)
+		const value = int64(10)
+
+		var signers = []addr.Address{anne, bob}
+		var nilParams = abi.MethodParams{}
+
+		rt := builder.WithBalance(abi.NewTokenAmount(0), abi.NewTokenAmount(0)).Build()
+
+		actor.mustConstructMultisigActor(rt, numApprovals, unlockDuration, signers...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.mustProposeWithExitCode(rt, chuck, value, method, nilParams, exitcode.ErrInsufficientFunds)
```

Comment:
To change in a follow-up: I think this should succeed, but the return value should contain the exit code of the nested send failure. This would be consistent with behaviour when N>1. But discuss with Lotus team first.